hack/test-end-to-end-docker.sh

```diff
@@ -66,11 +76,16 @@ function cleanup()
     echo "[INFO] Stopping k8s docker containers"; docker ps | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker stop
     if [[ -z "${SKIP_IMAGE_CLEANUP-}" ]]; then
       echo "[INFO] Removing k8s docker containers"; docker ps -a | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker rm
+
+      # clean up any residual volume mounts
+      mount | grep pods | grep "${VOLUME_DIR}" | awk '{print $3}' | sudo xargs -r -n 1 umount
```

Comment:
I'm tempted to run `oc delete all --all` for the test and default projects, and then run a `wait_for_command` that waits for the # of mounts in `VOLUME_DIR` to be 0.

Also, while we can do this before we clean the temp dir, the issue I'm facing is that e2e-docker runs and creates mounts in /tmp/openshift-e2e/containerized/volumes, finishes running, and then e2e starts and tries to `rm -rf` /tmp/openshift-e2e. This fails because there are some lingering mounts from the e2e-docker run.